includes/compat/trait-wc-stripe-subscriptions.php

```
@@ -742,8 +732,60 @@ public function process_subscription_payment( $amount, $renewal_order, $retry =
+	 * @param float    $amount         The amount to charge.
+	 * @param WC_Order $renewal_order  The renewal order.
+	 * @param object   $prepared_source The prepared renewal order source.
+	 * @return array{response: stdClass, is_authentication_required: bool}|null
+	 * @throws WC_Stripe_Exception When the Stripe API request fails.
+	 */
+	private function create_subscription_renewal_payment_response( $amount, WC_Order $renewal_order, $prepared_source ): ?array {
+		$order_helper = WC_Stripe_Order_Helper::get_instance();
+		$order_id     = $renewal_order->get_id();
+
+		if ( $order_helper->lock_order_payment( $renewal_order ) ) {
+			WC_Stripe_Logger::warning( "Stripe: skipping duplicate renewal attempt for order {$order_id} because the payment lock is already held." );
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'WC_Stripe_Logger' in method 'create_subscription_renewal_payment_response'. (undefined)

(StaticAccess)